src/plugins/mini/mini.c

```diff
@@ -0,0 +1,218 @@
+	char * before = NULL;
+
+	while (*equals != '\0' && (*equals != '=' || (before && *before == '\\')))
+	{
+		before = equals++;
+	}
+	return equals;
+}
+
+static inline char * unescape (char const * const text)
+{
+	return elektraReplace (text, "\\=", "=");
+}
+
+static inline void parseLine (char * line, size_t lineNumber, KeySet * keySet, Key * parentKey)
+{
+	char * pair = elektraStrip (stripComment (line));
+
+	if (*pair == '\0')
+	{
+		return;
+	}
+
+	char * equals = findUnescapedEquals (pair);
+	if (*equals == '\0' || equals == pair)
+	{
+		fprintf (stderr, "Ignored line %lu since “%s” does not contain a valid key value pair\n", lineNumber, pair);
+		ELEKTRA_ADD_WARNINGF (175, parentKey, "Line %lu: “%s” is not a valid key value pair", lineNumber, pair);
+		return;
+	}
+
+	*equals = '\0';
+
+	char * name = unescape (elektraRstrip (pair, NULL));
+	char * value = unescape (elektraLskip (equals + 1));
+
+	Key * key = keyNew (keyName (parentKey), KEY_END);
+	keyAddName (key, name);
+	keySetString (key, value);
+	ELEKTRA_LOG_DEBUG ("Name:  “%s”", keyName (key));
+	ELEKTRA_LOG_DEBUG ("Value: “%s”", keyString (key));
+
+	free (name);
+	free (value);
+
+	ksAppendKey (keySet, key);
+}
+
+static int parseINI (FILE * file, KeySet * keySet, Key * parentKey)
+{
+	char * line = NULL;
+	size_t capacity = 0;
+	int errorNumber = errno;
+
+	for (size_t lineNumber = 1; getline (&line, &capacity, file) != -1; ++lineNumber)
+	{
+		ELEKTRA_LOG_DEBUG ("Read Line %lu: %s", lineNumber, line);
+		parseLine (line, lineNumber, keySet, parentKey);
+	}
+
+	elektraFree (line);
+
+	if (!feof (file))
+	{
+		ELEKTRA_LOG_WARNING ("Did not reach end of configuration file “%s”", keyString (parentKey));
+		ELEKTRA_SET_ERROR (ELEKTRA_ERROR_NOEOF, parentKey, strerror (errno));
+		errno = errorNumber;
+		return ELEKTRA_PLUGIN_STATUS_ERROR;
+	}
+
+	return ELEKTRA_PLUGIN_STATUS_SUCCESS;
+}
+
+static int parseFile (KeySet * returned ELEKTRA_UNUSED, Key * parentKey)
+{
+	ELEKTRA_LOG ("Read configuration data");
+	int errorNumber = errno;
+	FILE * source = fopen (keyString (parentKey), "r");
+
+	if (!source)
+	{
+		ELEKTRA_LOG_WARNING ("Could not open file “%s” for reading: %s", keyString (parentKey), strerror (errno));
+		ELEKTRA_SET_ERROR_GET (parentKey);
+		errno = errorNumber;
+		return ELEKTRA_PLUGIN_STATUS_ERROR;
+	}
+
+	int status = parseINI (source, returned, parentKey);
+
+	fclose (source);
+	return status;
+}
+
+static inline char * escape (char const * const text)
+{
+	return elektraReplace (text, "=", "\\=");
+}
+
+static inline void writeFile (FILE * file, KeySet * keySet, Key * parentKey)
+{
+	Key * key;
+	ksRewind (keySet);
+	while ((key = ksNext (keySet)) != 0)
+	{
+		const char * name = elektraKeyGetRelativeName (key, parentKey);
+		ELEKTRA_LOG_DEBUG ("Write mapping “%s=%s”", name, keyString (key));
+
+		char * escapedName = escape (name);
+		char * escapedKey = escape (keyString (key));
+		fprintf (file, "%s=%s\n", escapedName, escapedKey);
+		free (escapedName);
```

Comment:
It is enough if you do :+1: for comments that were addressed and/or mark as "ready to merge" if all issues were addressed (or created as separate issues to be resolved later).